### ISSUE TITLE
Refresh Event Modal

### ISSUE THREAD

**Issue by Caleb-Cohen** [Together-100Devs/Together#285]
## Please provide a summary of the feature
@timemir has offered to make more significant updates to the event view modal than was originally planned for this issue so I've updated the issue to contain more information.

Task:
- Update the event view modal to have the description on the outside of the Description box
- Move the delete buttons to the bottom
- Add to calendar button*
- Add RSVP button*
- Add Edit button*

*Buttons should be hideable so redesign does not get held up by finishing the necessary features related to each button. This makes it so we do not need to potentially redesign the event view to add more buttons that were not originally considered.

I've included rough draft @timemir has worked on. Pending final design approval and changes by @Natashka333.

<img width="502" alt="215829229-3772cc33-155f-4417-94a7-9733a46a69e4" src="https://user-images.githubusercontent.com/75268174/215944136-9f929b2c-09c6-4990-9295-1dd2d59c4051.png">

## Are there any potential issues you foresee with this feature
With any redesign, especially ones that take a bit of time, merge conflicts will occur and need to be dealt with. Developers should be referred to this issue if they are working on closely related material.

## Which Branch should this fix be Pulled into?
PR into eventModalRefresh

*If you're interested in working on this feature, please comment so you can be assigned*

**Comment by Natashka333** [comment]
![Screen Shot 2023-01-24 at 4 33 41 PM](https://user-images.githubusercontent.com/102708144/214436313-04258d7d-434f-4d6d-af12-ba27356699cd.png)
@Caleb-Cohen what do we think about this?

**Comment by Caleb-Cohen** [comment]
> ![Screen Shot 2023-01-24 at 4 33 41 PM](https://user-images.githubusercontent.com/102708144/214436313-04258d7d-434f-4d6d-af12-ba27356699cd.png)
> 
> @Caleb-Cohen what do we think about this?

This is quite different than we already have I do like the buttons though. Why is the X on the right side?

**Comment by Natashka333** [comment]
@Caleb-Cohen I used RSVP, and trash can buttons from mui. Obviously, we can use the same icon for them in this window. 
The X was on the editing window. But I can move to the upper left corner.

**Comment by Caleb-Cohen** [comment]
> Hey, I would love to be assigned to this. However, I would like to expand the issue to redesign the whole modal, if that is okay with you. (kukli from discord)

Hey there! Welcome to the Together team! I've assigned you the issue.

<img width="818" alt="Bildschirmfoto_2023-01-31_um_16 27 11" src="https://user-images.githubusercontent.com/75268174/215813898-cfbbefa2-6c16-4dff-b1a0-a9e2655dda44.png">

This looks great! That being said, this does expand the scope original issue. @Natashka333 is the lead designer, and design changes should be viewed and cleared before implementation.

**Comment by timemir** [comment]
Hey, thanks!

This image also shows the confirmation area that expands when a delete button is pressed:

<img width="502" alt="Bildschirmfoto 2023-01-31 um 17 53 19" src="https://user-images.githubusercontent.com/84287747/215829229-3772cc33-155f-4417-94a7-9733a46a69e4.png">

**Comment by Natashka333** [comment]
> Hey, thanks!
> 
> 
> 
> This image also shows the confirmation area that expands when a delete button is pressed:
> 
> 
> 
> <img width="502" alt="Bildschirmfoto 2023-01-31 um 17 53 19" src="https://user-images.githubusercontent.com/84287747/215829229-3772cc33-155f-4417-94a7-9733a46a69e4.png">
> 
> 

Hey Tim! Welcome to Together!

I love some of the changes you’ve made to the current modal. Specifically, the look of the date and time and moving the delete buttons to the bottom for admins. 

There will need to be some changes if you’re wanting to take on a redesign. We’ll need to add a RSVP button, a delete button for authors of an event, and editing for even authors. Additionally, some colors will need to be changed to more closely fit Together’s style. 

I will be back home tomorrow and will throw this design into Figma to get this process started. Are you familiar with Figma?

Really great work, thanks for contributing!

**Comment by timemir** [comment]
> 

Thank you so much! Yes, I am very familiar with Figma, but I designed this in code, using Tailwind. It was just fun to play around with the code. Anyway, yeah, it would be a lot smarter to just go ahead with Figma! :)

The section below is only shown when "authorCheck" is true, so this represents the delete buttons for the author. This could be renamed to "author control panel" or something if it is confusing. 

Being able to edit the post would involve a lot of new logic, including in the backend, and perhaps it would be better to do this in a whole separate issue? However, adding just a visual button should be no problem!

I am not familiar with what an RSVP button is.

**Comment by Natashka333** [comment]
> > 
> 
> 
> 
> Thank you so much! Yes, I am very familiar with Figma, but I designed this in code, using Tailwind. It was just fun to play around with the code. Anyway, yeah, it would be a lot smarter to just go ahead with Figma! :)
> 
> 
> 
> The section below is only shown when "authorCheck" is true, so this represents the delete buttons for the author. This could be renamed to "author control panel" or something if it is confusing. 
> 
> 
> 
> Being able to edit the post would involve a lot of new logic, including in the backend, and perhaps it would be better to do this in a whole separate issue? However, adding just a visual button should be no problem!
> 
> 
> 
> I am not familiar with what an RSVP button is.



Cool! I would be happy to work with you. Let me get it in Figma and we can schedule time to talk?

**Comment by Caleb-Cohen** [comment]
> Being able to edit the post would involve a lot of new logic, including in the backend, and perhaps it would be better to do this in a whole separate issue? However, adding just a visual button should be no problem!

I agree. For this issue I'd like to get an edit button put somewhere so that we have a dedicated spot and don't need to do the design again. Button will be hidden until functionality has been finished.

RSVP would be for users to subscribe to events so that they can be notified when events they are subscribed to are starting. This would follow the same philosophy as the edit button. We want a place to put it, but it can be hidden until the functionality is fully ready.

**Comment by intelagense** [comment]
Since the RSVP is not very formal or even required to attend, it might be better to name it something like "Attending," "Interested," or "Join."

**Comment by Caleb-Cohen** [comment]
> Since the RSVP is not very formal or even required to attend, it might be better to name it something like "Attending," "Interested," or "Join."

Good point.

**Comment by intelagense** [comment]
Also I want to note here that the design @timemir suggested can automagically resolve #307

**Comment by timemir** [comment]
> > > 
> > 
> > 
> > Thank you so much! Yes, I am very familiar with Figma, but I designed this in code, using Tailwind. It was just fun to play around with the code. Anyway, yeah, it would be a lot smarter to just go ahead with Figma! :)
> > The section below is only shown when "authorCheck" is true, so this represents the delete buttons for the author. This could be renamed to "author control panel" or something if it is confusing.
> > Being able to edit the post would involve a lot of new logic, including in the backend, and perhaps it would be better to do this in a whole separate issue? However, adding just a visual button should be no problem!
> > I am not familiar with what an RSVP button is.
> 
> Cool! I would be happy to work with you. Let me get it in Figma and we can schedule time to talk?

Of course! Let us get in touch on Discord (kukli#4752). I am based in Germany, so sorry in advance if my replies are a bit delayed! :)

**Comment by Natashka333** [comment]
@timemir, @Caleb-Cohen, and everyone, I have tried to bring Tim's redesign close to Together's design. 
![Screen Shot 2023-02-02 at 11 31 30 PM](https://user-images.githubusercontent.com/102708144/216520385-1c7c50e8-f5e9-48af-a805-1dfb6c93349e.png)
![Screen Shot 2023-02-02 at 11 31 41 PM](https://user-images.githubusercontent.com/102708144/216520392-90ee452e-bf2d-4489-8858-7a9b91b6a930.png)

**Comment by Caleb-Cohen** [comment]
> @timemir, @Caleb-Cohen, and everyone, I have tried to bring Tim's redesign close to Together's design. ![Screen Shot 2023-02-02 at 11 31 30 PM](https://user-images.githubusercontent.com/102708144/216520385-1c7c50e8-f5e9-48af-a805-1dfb6c93349e.png) ![Screen Shot 2023-02-02 at 11 31 41 PM](https://user-images.githubusercontent.com/102708144/216520392-90ee452e-bf2d-4489-8858-7a9b91b6a930.png)

Good use of the white space. I think the red outline for the delete events should be the orange that Together has in it's color palette.

**Comment by Natashka333** [comment]
![Screen Shot 2023-02-02 at 11 34 55 PM](https://user-images.githubusercontent.com/102708144/216520890-28a503d2-826f-455c-ae89-36f97c039998.png)

**Comment by timemir** [comment]
I also like that you put the location and author next to each other! 

However, I would like to suggest some changes and talk about the design in general. Should we talk on Discord or should I write it out here? :D

**Comment by Caleb-Cohen** [comment]
Howdy @timemir and @Natashka333 - just wanted to check if y'all are still interested in working on this? If not, we can open it up to another individual.

**Comment by Natashka333** [comment]
> Howdy @timemir and @Natashka333 - just wanted to check if y'all are still interested in working on this? If not, we can open it up to another individual. 

@Caleb-Cohen I'm still interested 😉

**Comment by timemir** [comment]
> Howdy @timemir and @Natashka333 - just wanted to check if y'all are still interested in working on this? If not, we can open it up to another individual.

Hey, I am really sorry for not getting back to you sooner! I am currently extremely busy writing my master's thesis and working on a new job. Unfortunately, I do not have time for together in the near future.

**Comment by Caleb-Cohen** [comment]
> > Howdy @timemir and @Natashka333 - just wanted to check if y'all are still interested in working on this? If not, we can open it up to another individual.
> 
> Hey, I am really sorry for not getting back to you sooner! I am currently extremely busy writing my master's thesis and working on a new job. Unfortunately, I do not have time for together in the near future.

Hey Tim! We understand, glad to hear the master's thesis and job are keeping you busy.

I'll re-open the issue to other individuals. Do you have any of the earlier mockups in tailwind that we can continue the work off of?

**Comment by timemir** [comment]
> > > Howdy @timemir and @Natashka333 - just wanted to check if y'all are still interested in working on this? If not, we can open it up to another individual.
> > 
> > 
> > Hey, I am really sorry for not getting back to you sooner! I am currently extremely busy writing my master's thesis and working on a new job. Unfortunately, I do not have time for together in the near future.
> 
> Hey Tim! We understand, glad to hear the master's thesis and job are keeping you busy.
> 
> I'll re-open the issue to other individuals. Do you have any of the earlier mockups in tailwind that we can continue the work off of?

Thanks for understanding. :) 
Here is the old code for the EventModal that I have: https://gist.github.com/timemir/13ee28f4f6fa98fa981ca1a1e56d11df

**Comment by Caleb-Cohen** [comment]
@Natashka333 this may be a great issue for you to take on to learn how to work with tailwind and react. I would recommend pair programming with one of our developers that really know this stuff well if that's something you want to pursue.

**Comment by Natashka333** [comment]
Sure, I would love to try.

**Comment by luciousmc** [comment]
I can help out with this issue if you want @Natashka333 @Caleb-Cohen

**Comment by Natashka333** [comment]
@luciousmc I am pretty new to Tailwind and React.  Would you like to work on it together?

**Comment by luciousmc** [comment]
Definitely,  we can pair program on it. We can coordinate on discord when to do it.

Don't worry about being new to React and tailwind, I'll help you through it. 👍

**Comment by Natashka333** [comment]
Yay! Let's do it!

**Comment by Caleb-Cohen** [comment]
> @luciousmc Yay! Let's do it!

should I close this issue out?

**Comment by Caleb-Cohen** [comment]
Closed / Lack of Interest
